components/gantry/singleaxis/singleaxis.go

```diff
@@ -22,21 +22,24 @@ import (
 	spatial "go.viam.com/rdk/spatialmath"
 )
 
-var model = resource.DefaultModelFamily.WithModel("oneaxis")
+var (
+	model = resource.DefaultModelFamily.WithModel("single-axis")
+	wg    sync.WaitGroup
+)
 
 // limitErrorMargin is added or subtracted from the location of the limit switch to ensure the switch is not passed.
 const limitErrorMargin = 0.25
 
-// Config is used for converting oneAxis config attributes.
+// Config is used for converting singleAxis config attributes.
 type Config struct {
-	Board           string    `json:"board,omitempty"` // used to read limit switch pins and control motor with gpio pins
-	Motor           string    `json:"motor"`
-	LimitSwitchPins []string  `json:"limit_pins,omitempty"`
-	LimitPinEnabled *bool     `json:"limit_pin_enabled_high,omitempty"`
-	LengthMm        float64   `json:"length_mm"`
-	MmPerRevolution float64   `json:"mm_per_rev,omitempty"`
-	GantryRPM       float64   `json:"gantry_rpm,omitempty"`
-	Axis            r3.Vector `json:"axis"`
+	Board           string   `json:"board,omitempty"` // used to read limit switch pins and control motor with gpio pins
+	Motor           string   `json:"motor"`
+	LimitSwitchPins []string `json:"limit_pins,omitempty"`
+	LimitPinEnabled *bool    `json:"limit_pin_enabled_high,omitempty"`
+	LengthMm        float64  `json:"length_mm"`
+	MmPerRevolution float64  `json:"mm_per_rev"`
+	GantryMmPerSec  float64  `json:"gantry_mm_per_sec,omitempty"`
```

Comment:
This needs to be verified in Validate.

Reply:
🤔 So, if I set GantryMmPerSec to 0 what is the expected outcome?

Reply:
I thought we were handling GantryRPM -> GantryMMPerSec in a separate pr? Could we do that, because that needs additional motorRPM = GantryMMPerSec*mmPerRev*convertToRPM for speeds.

Reply:
I didn't know we were handling them separately. I thought you just wanted both for now but yeah I'll revert to ONLY gantry rpm for now

Reply:
@susmitaSanyal I would assume the expected behavior if you set `GantryMmPerSec` to 0 is for the gantry to not move because it will "go" 0 millimeters every second. Not applicable to this PR I guess so we'll have to clarify the behavior more when the time comes.

Reply:
Sorry for the lack of clarity on my part!

Reply:
all good! I'll make another ticket for that issue